src/_pytest/python.py

```diff
@@ -1016,6 +1042,7 @@ def parametrize(self, argnames, argvalues, indirect=False, ids=None, scope=None)
                 )
                 newcalls.append(newcallspec)
         self._calls = newcalls
+        return ids
```

Comment:
Not happy about having to return this here (for storing it in `pytest_generate_tests` then).

Reply:
Solving the wrong problem at the wrong place feels like that

Reply:
The main problem here is passing `args` and `kwargs`, but not the mark.
This could be done instead here also - but the function can be used by itself also (without a mark), so I wanted to keep it apart.  But having an optional `store_resolved_ids_on_mark` kwarg might suffice then.

Returning the generated ids is not too bad also after all, but I've felt that it might be limiting in the future, and should therefore return the generated calls also etc.